IssueService.java

```diff
@@ -0,0 +1,57 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class IssueService {
+
+	@Autowired
+	private IssueRepository issueRepository;
+
+	public Issue update(Issue issue) throws InvalidIssueStatusException {
+
+		// TODO: Validate the issue.status first;
+		// Status can be updated from CREATED to IN_PROGRESS to COMPLETED.
+		if(!isValidStatus(issue)){
+			throw new InvalidIssueStatusException("Issue status is invalid!");
+		}
+		
+		// TODO: Save the issue using issueRepository
+		return issueRepository.updateIssue(issue);
+	}
+	
+	/**
+	 * Checks if issue status is valid to update
+	 * 
+	 * @param issue
+	 * @return
+	 */
+	private boolean isValidStatus(Issue issue) {
+		try {
+			if (issue == null
```

Comment:
if issue is null, we are throwing `InvalidIssueStatusException` from `update` method? I think this null condition should go into `update` method only

Reply:
first step is validation and then update, if issue is null why we will call update?